src/Compilers/Test/Utilities/CSharp.Desktop/CSharpTestBase.cs

```diff
@@ -253,7 +253,16 @@ public static SyntaxTree Parse(string text, string filename = "", CSharpParseOpt
             }
 
             var stringText = StringText.From(text, Encoding.UTF8);
-            return SyntaxFactory.ParseSyntaxTree(stringText, options, filename);
+            return CheckSerializable(SyntaxFactory.ParseSyntaxTree(stringText, options, filename));
+        }
+
+        private static SyntaxTree CheckSerializable(SyntaxTree tree)
+        {
+            var stream = new MemoryStream();
+            tree.GetRoot().SerializeTo(stream);
+            stream.Position = 0;
+            var deserialized = CSharpSyntaxNode.DeserializeFrom(stream);
+            return tree;
```

Comment:
Comparing nodes is also recursive.  We'll need to fix that in a separate PR.

---
In reply to: [84809576](https://github.com/dotnet/roslyn/pull/14671#discussion_r84809576) [](ancestors = 84809576)